google/cloud/aiplatform/models.py

```diff
@@ -961,8 +995,12 @@ def predict(self, instances: List, parameters: Optional[Dict] = None) -> Predict
 
         """
         self.wait()
-
-        prediction_response = self._prediction_client.predict(
+        self._prediction_api_client = self._instantiate_prediction_client(
```

Comment:
Similar to the above comment about treating the `ClientWithOverride` directly as a client we wouldn't need to instantiate the client we can just directly call predict.